.github/workflows/ci.yml

```
@@ -110,7 +110,7 @@ jobs:
   test:
     strategy:
       matrix:
-        environment: [ubuntu-latest, macos-latest, windows-latest]
+        environment: [ubuntu-latest, macos-latest, macos-14, windows-latest]
```

```suggestion
        environment: [ubuntu-latest, macos-12, macos-14, windows-latest]
```

We should pin `macos-latest` to either 12 or 13, so that it stays being x86 for the nearby future. Apparently the plan is to change `macos-latest` to mean `macos-14` in April 2024 and we would end up with two arm64 builds 😅 

https://github.blog/changelog/2024-01-30-github-actions-macos-14-sonoma-is-now-available/

https://github.com/actions/runner-images?tab=readme-ov-file#available-images